### PR TITLE
cmd/systray: improve profile menu

### PR DESCRIPTION
**Pull request by willnorris**
Bring UI closer to macOS and windows:
- split login and tailnet name over separate lines
- render profile picture (with very simple caching)
- use checkbox to indicate active profile. I've not found any desktops that can't render checkboxes, so I'd like to explore other options if needed.

before and after:
profile-before
profile-after

Updates #1708

Change-Id: Ia101a4a3005adb9118051b3416f5a64a4a45987d

### CODE REVIEW

**Comment by tendstofortytwo** [comment]
okay, I actually can't test this because of https://github.com/tailscale/tailscale/issues/14431

:see_no_evil:

File: cmd/systray/systray.go [hunk 1]
```diff
@@ -119,9 +120,13 @@ func (menu *Menu) rebuild(state state) {
 
 	account := "Account"
 	if state.curProfile.Name != "" {
-		account += fmt.Sprintf(" (%s)", state.curProfile.Name)
+		account = state.curProfile.Name
+		if state.curProfile.NetworkProfile.DomainName != "" {
+			account += "\n" + state.curProfile.NetworkProfile.DomainName
+		}
```
Comment (bradfitz) [inline cmd/systray/systray.go:126]:
```go
account = state.curProfile.Name + condNewline(state.curProfile.NetworkProfile.DomainName)

func condNewline(s string) string {
    if s == "" {
        return ""
    }
    return "\n" + s
}
```

maybe?

File: cmd/systray/systray.go [hunk 2]
```diff
@@ -132,13 +137,17 @@ func (menu *Menu) rebuild(state state) {
 	// Aggregate all clicks into a shared channel.
 	menu.accountsCh = make(chan ipn.ProfileID)
 	for _, profile := range state.allProfiles {
-		title := fmt.Sprintf("%s (%s)", profile.Name, profile.NetworkProfile.DomainName)
-		// Note: we could use AddSubMenuItemCheckbox instead of this formatting
-		// hack, but checkboxes don't work across all desktops unfortunately.
+		title := profile.Name
```
Comment (bradfitz) [inline cmd/systray/systray.go:140]: `title := profile.Name + condNewline(profile.NetworkProfile.DomainName)` again?

File: cmd/systray/systray.go [hunk 3]
```diff
@@ -170,6 +179,34 @@ func (menu *Menu) rebuild(state state) {
 	go menu.eventLoop(ctx)
 }
 
+var (
+	cacheMu   sync.Mutex
+	httpCache = map[string][]byte{}
```
Comment (bradfitz) [inline cmd/systray/systray.go:184]: `= map[string][]byte{} // keyed by foo`

or `// foo => bar`

File: cmd/systray/systray.go [hunk 4]
```diff
@@ -170,6 +179,34 @@ func (menu *Menu) rebuild(state state) {
...
+func setRemoteIcon(menu *systray.MenuItem, urlStr string) {
+	if menu == nil || urlStr == "" {
+		return
+	}
+
+	cacheMu.Lock()
+	b, ok := httpCache[urlStr]
+	if !ok {
+		if resp, err := http.Get(urlStr); err == nil {
+			b, _ = io.ReadAll(resp.Body)
```
Comment (bradfitz) [inline cmd/systray/systray.go:198]: ignoring resp.StatusCode?

**Review by bradfitz — APPROVED** [review]

File: cmd/systray/systray.go (hunk 1 above)
Comment (willnorris) [reply cmd/systray/systray.go:126]: I pulled it out to a `profileTitle` func, which was something I had originally but ended up removing.

File: cmd/systray/systray.go (hunk 3 above)
Comment (willnorris) [reply cmd/systray/systray.go:184]: done

File: cmd/systray/systray.go (hunk 4 above)
Comment (willnorris) [reply cmd/systray/systray.go:198]: fixed

**Pull request merged**